pkg/controllers/testing/factories/keda_scaledobject.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2020 the original author or authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package factories
+
+import (
+	"fmt"
+
+	kedav1alpha1 "github.com/projectriff/system/pkg/apis/thirdparty/keda/v1alpha1"
+)
+
+type kedaScaledObject struct {
+	target *kedav1alpha1.ScaledObject
+}
+
+func KedaScaledObject(seed ...*kedav1alpha1.ScaledObject) *kedaScaledObject {
+	var target *kedav1alpha1.ScaledObject
+	switch len(seed) {
+	case 0:
+		target = &kedav1alpha1.ScaledObject{}
+	case 1:
+		target = seed[0]
+	default:
+		panic(fmt.Errorf("expected exactly zero or one seed, got %v", seed))
+	}
+	return &kedaScaledObject{
+		target: target,
+	}
+}
+
+func (p *kedaScaledObject) deepCopy() *kedaScaledObject {
```

Comment:
nit: the receiver name is inconsistent. For the other factories, I've used `f`

Reply:
Changed to `f` throughout.